common/ustr.h

```diff
@@ -52,66 +53,24 @@ class UnicodeBiDiText;
  * The presence of \0 characters in the string will cause undefined
  * behavior in some operations.
  */
-class U32String {
+class U32String : public BaseString<uint32> {
```

Comment:
Maybe use `char32_t` instead of `uint32` when `USE_CXX11` is defined to be consistent with UTF-32 string literal in C++11. That would also allow creating a `U32String` from a `U"C++11 UTF-32 string literal"`.

An alternative to this would be to keep using `uint32` but at least add a constructor that takes a `const char32_t *` so that we can create a `U32String` from a UTF-32 string literal.

Reply:
Done